packages/utils/src/bulk-insert-patients.ts

```diff
@@ -140,25 +149,94 @@ async function loadData(
   console.log(`Done, inserted ${successfulCount} patients.`);
 }
 
-async function displayWarningAndConfirmation(results: unknown[], orgName: string, dryRun: boolean) {
+function dedupPatientCreates(patients: PatientCreate[]): PatientCreate[] {
+  const patientMap = new Map<string, PatientCreate>();
+  patients.forEach(patient => {
+    const nameKey = `${patient.firstName} ${patient.lastName}`;
+    const existing = patientMap.get(nameKey);
+    if (existing) {
+      const mergedPatient = mergePatients(existing, patient);
+      patientMap.set(nameKey, mergedPatient);
+    } else {
+      patientMap.set(nameKey, patient);
+    }
+  });
+  return Array.from(patientMap.values());
+}
+
+function mergePatients(p1: PatientCreate, p2: PatientCreate): PatientCreate {
+  const addresses = [
+    ...(Array.isArray(p1.address) ? p1.address : [p1.address]),
+    ...(Array.isArray(p2.address) ? p2.address : [p2.address]),
+  ];
+  const uniqueAddresses = deduplicateAddresses(addresses);
+
+  const contacts = [
+    ...(Array.isArray(p1.contact) ? p1.contact : p1.contact ? [p1.contact] : []),
+    ...(Array.isArray(p2.contact) ? p2.contact : p2.contact ? [p2.contact] : []),
+  ];
+  const uniqueContacts = deduplicateContacts(contacts);
+
+  return {
+    ...p1,
+    address: [uniqueAddresses[0], ...uniqueAddresses.slice(1)],
```

Comment:
i assume this is done to prevent shallow copying? can't we use clearer methods for it, i.e. cloneDeep? or is this common/standard practice?

Reply:
This is to make the compiler happy. The type of `address` requires at least one item on the array.